src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -812,8 +827,185 @@ def __set_errors_json(self, error_count_by_operation, errors_by_operation):
+                            __get_errors_from_substatus()
+                            __recompose_truncated_substatus_msg()
+                                __recompose_substatus_msg_errors()
+                                __create_installation_tombstone
+                                __recreate_installation_summary_json()
+                                
+                        status_file_size_in_bytes, status_file_agent_size_diff = __get_new_size_in_bytes_after_truncation(truncated_status_file)
+                        size_of_max_packages_allowed_in_status -= status_file_agent_size_diff  
+        """
+        self.composite_logger.log_debug("Begin package list truncation")
+        truncated_status_file = json.loads(complete_status_file_payload_json)  # reload payload into python object
+        low_pri_index = None
+        assessment_substatus_index = self.__get_substatus_index(Constants.PATCH_ASSESSMENT_SUMMARY, truncated_status_file['status']['substatus'])
+        installation_substatus_index = self.__get_substatus_index(Constants.PATCH_INSTALLATION_SUMMARY, truncated_status_file['status']['substatus'])
+
+        if assessment_substatus_index is not None:      # If assessment data exists
+            self.__assessment_substatus_msg_copy = self.__get_substatus_message(truncated_status_file, assessment_substatus_index)
+            self.__assessment_packages_copy = self.__assessment_substatus_msg_copy['patches']
+
+        if installation_substatus_index is not None:    # If installation data exists
+            self.__installation_substatus_msg_copy = self.__get_substatus_message(truncated_status_file, installation_substatus_index)
+            self.__installation_packages_copy = self.__installation_substatus_msg_copy['patches']
+            low_pri_index = self.__get_installation_low_pri_index(self.__installation_packages_copy)
+
+        status_file_without_package_list_size = self.size_of_constant_status_data(copy.deepcopy(truncated_status_file), assessment_substatus_index, installation_substatus_index)  # Deepcopy, fully copies the object to avoid reference modifications
+        size_of_max_packages_allowed_in_status = Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES - status_file_without_package_list_size
+
+        while status_file_size_in_bytes > Constants.StatusTruncationConfig.INTERNAL_FILE_SIZE_LIMIT_IN_BYTES:
+            # Start truncation process
+            packages_retained_in_assessment, packages_removed_from_assessment, packages_retained_in_installation, packages_removed_from_installation = \
+                self.__apply_truncation_process(self.__assessment_packages_copy, self.__installation_packages_copy, size_of_max_packages_allowed_in_status, low_pri_index)
+
+            if len(packages_removed_from_assessment) > 0:
+                self.composite_logger.log_debug("Recompose truncated status file json payload: {0}".format(Constants.ASSESSMENT))
```

Comment:
self.composite_logger.log_debug("Recomposing truncated status payload for [Substatus={0}]".format(Constants.PATCH_ASSESSMENT_SUMMARY)) 

Constants.ASSESSMENT is the overall patch operation. 

Individual substatuses within the status file, that are being modified here, are PatchAssessmentSummary and PatchInstallationSummary.

Same change applies installation section on line 922

Reply:
done

Reply:
nit: Recomposing truncated status payload **for** [Substatus={0}]

Applies for line 901 also

Reply:
@feng-j678 This is still pending